### PR TITLE
Reduce Debug Module "flags"

### PR DESCRIPTION
**Pull request by mwachs5**
This PR does a few things:

1) Reduces the number of zero, read-only `flag` registers in the Debug Module.
2) Removes the `[X:Y]` notation from register names generated with the `RegField.bytes` function, since it doesn't really make sense to have those characters in the name of a register. Instead, they are simply indexed by `_Y` where `Y` is the byte number
3) Prevents the generated JSON files from overwriting eachother if they are actually referring to different address spaces.

### CODE REVIEW

File: src/main/scala/tilelink/RegisterRouter.scala [hunk 1]
```diff
@@ -111,7 +111,12 @@ case class TLRegisterNode(
       ("baseAddress" -> base) ~
       ("regfields" -> regDescs)
     ))
-    ElaborationArtefacts.add(s"${base}.regmap.json", pretty(render(json)))
+
+    var suffix = 0
+    while( ElaborationArtefacts.contains(s"${base}.${suffix}.regmap.json")){
+      suffix = suffix + 1
+    }
+    ElaborationArtefacts.add(s"${base}.${suffix}regmap.json", pretty(render(json)))
```
Comment (hcook) [inline src/main/scala/tilelink/RegisterRouter.scala]: `s"${base}.${suffix}**.**regmap.json"` != `s"${base}.${suffix}regmap.json"` one has a period and one doesnt

File: src/main/scala/util/GeneratorUtils.scala [hunk 2]
```diff
@@ -139,7 +139,12 @@ trait GeneratorApp extends App with HasGeneratorUtilities {
 
 object ElaborationArtefacts {
   var files: Seq[(String, () => String)] = Nil
-  def add(extension: String, contents: => String) {
+
+  def add(extension: String, contents: => String) = {
```
Comment (hcook) [inline src/main/scala/util/GeneratorUtils.scala]: Are you intentionally making this no longer a `Unit` function with `=`?

File: src/main/scala/tilelink/RegisterRouter.scala (hunk 1 above)
Comment (mwachs5) [reply src/main/scala/tilelink/RegisterRouter.scala]: thanks!

File: src/main/scala/util/GeneratorUtils.scala (hunk 2 above)
Comment (mwachs5) [reply src/main/scala/util/GeneratorUtils.scala]: no. thanks.

**Pull request merged**